lib/iris/_lazy_data.py

```diff
@@ -201,6 +201,25 @@ def multidim_lazy_stack(stack):
     return result
 
 
+def lazy_concat(data, axis):
```

Comment:
Not terribly convinced about the usefulness of this ...
I'm opposed to duplicating APIs unless you are really defining a complete abstraction layer with its own value.
-- for example, what if da.concatenate later acquires useful keywords, which then have to be added here...

In this case, I don't think abstracting *all* usage dask is obvioiusly helpful in itself, so I don't see a great need for this specific function.
So I think I would ditch this, just to make the PR tighter.